pkg/credentials/utils.go

```diff
@@ -0,0 +1,32 @@
+package credentials
+
+import (
+	"encoding/base64"
+	"strings"
+)
+
+func ToBase64Url(buffer []byte) string {
+	encoded := base64.RawURLEncoding.EncodeToString(buffer)
```

Comment:
this is already base64url encoding (without padding), you don't need to do the replacing.